docker-compose.yml

```diff
@@ -54,8 +54,7 @@ services:
     volumes:
       - ./benchmarks/data:/data
     depends_on:
-      ballista-scheduler:
-        condition: service_healthy
```

Comment:
We should be able to start executors before the scheduler runs, but I haven't tested this in a long time. If we require the scheduler to be running first, then we should file a bug for that.

Reply:
I tested this and the executor fails if it cannot connect to the scheduler. In theory, this should be fine in an environment such as k8s because the pod will just keep restarting until it succeeds.

Reply:
The restart config option could be added to the compose file. https://docs.docker.com/compose/compose-file/#restart
```
restart:always
```

Reply:
I am not sure what to do about this. We could have a config option for having the executor itself retry connecting to the scheduler and use that config option when using docker compose.

Reply:
WDYT @avantgardnerio ?

Reply:
Executor crashes if it starts before the scheduler is fully running (which seems to be the default case for me) and never restarts.

Reply:
I am tempted to add a `--retry-timeout--seconds 10` option to the executor, which would fix this. Ballista is targeted at enterprise users, and they are often not on the latest distro, so would be nice for this to work out of the box for as many people as possible.

Reply:
> I am tempted to add a `--retry-timeout--seconds 10` option to the executor, which would fix this. Ballista is targeted at enterprise users, and they are often not on the latest distro, so would be nice for this to work out of the box for as many people as possible.

I was thinking of the same option given that the application does not seem to have a hard constraint to be readily available within milliseconds on startup. I second the part with having it working out of the box for different users.

Reply:
@andygrove - I am not too familiar with the codebase yet, but I saw that if the scheduler is killed while executor is running, the executor tries to reconnect forever. Once the scheduler is restarted, the executor is fine. Probably a silly question, but couldn't this behaviour be there from the start? Why does it first need to connect to an actual running instance of the scheduler, instead of keep trying the default/configured one until  it succeeds?

Reply:
`TaskRunnerPool` loops forever, polling schedulers to fetch tasks to run. This code opens a new connection to the scheduler each time, which is inefficient but does mean that it eventually recovers from scheduler downtime.

Reply:
I pushed a commit to add the new config. This fixes it for me.

```
ballista-executor_1   | 2022-10-15T15:44:09.490225Z  WARN main ThreadId(01) ballista_executor: Failed to connect to scheduler at http://ballista-scheduler:50050 (Could not connect to scheduler); retrying ...    
ballista-executor_1   | 2022-10-15T15:44:09.991817Z  INFO main ThreadId(01) ballista_executor: Connected to scheduler at http://ballista-scheduler:50050    
```

Reply:
> `TaskRunnerPool` loops forever, polling schedulers to fetch tasks to run. This code opens a new connection to the scheduler each time, which is inefficient but does mean that it eventually recovers from scheduler downtime.

I see, I thought it might be the case. Thanks for taking the time to explain!